### PR TITLE
chore: bump github.com/Azure/azure-sdk-for-go/sdk/storage/azblob

### PR DESCRIPTION
**Pull request by roelarents**
Bumps [github.com/Azure/azure-sdk-for-go/sdk/storage/azblob](https://github.com/Azure/azure-sdk-for-go) from 1.0.0 to 1.2.1.
- [Release notes](https://github.com/Azure/azure-sdk-for-go/releases)
- [Changelog](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/release.md)
- [Commits](https://github.com/Azure/azure-sdk-for-go/compare/sdk/storage/azblob/v1.0.0...sdk/storage/azblob/v1.2.1)

Fixes: n/a. I didn't create an issue.

### Motivation

Update azure-sdk-for-go/sdk/storage/azblob to the latest version.

Dependabot would normally do this, but it has been explicitly [disabled](https://github.com/argoproj/argo-workflows/pull/10509#issuecomment-1427081626) for this dep by @alexec. Perhaps it could/should be enabled again, since @brianloss manually updated to the 1.0 version.

After updating argo from v3.4.8 to v3.5.2 we ran into [this issue](https://github.com/Azure/azure-sdk-for-go/issues/20735) in the azure sdk with uploading artifacts with a particular size in which case the upload part size is not correctly calculated.

### Modifications

A version update in go.mod and .sum

Also tagged on the expansion of the debug message for `executeTemplate`, to help investigating for this (unrelated) [issue with the recently introduced recursion limit](https://github.com/argoproj/argo-workflows/issues/12162)

### Verification

Tested by:
* updating this dep in 3.5.2
* running the unit tests
* building the argoexec image locally
* running the same workflow that failed on it before

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -1821,7 +1821,7 @@ type executeTemplateOpts struct {
 // nodeName is the name to be used as the name of the node, and boundaryID indicates which template
 // boundary this node belongs to.
 func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string, orgTmpl wfv1.TemplateReferenceHolder, tmplCtx *templateresolution.Context, args wfv1.Arguments, opts *executeTemplateOpts) (*wfv1.NodeStatus, error) {
-	woc.log.Debugf("Evaluating node %s: template: %s, boundaryID: %s", nodeName, common.GetTemplateHolderString(orgTmpl), opts.boundaryID)
+	woc.log.Debugf("Evaluating node %s: template: %s, boundaryID: %s, currentStackDepth: %v", nodeName, common.GetTemplateHolderString(orgTmpl), opts.boundaryID, woc.currentStackDepth)
```
Comment (terrytangyuan) [inline workflow/controller/operator.go:1824]: Remove unrelated change?

Comment (roelarents) [reply workflow/controller/operator.go]: ok

**Review by terrytangyuan — APPROVED** [review]

**Comment by agilgur5** [comment]
Backported cleanly into `release-3.5` as 5c596423b5b1ed2dc91d84e063f89f8846efb820
